src/storage/sql_table.cpp

```diff
@@ -23,8 +23,14 @@ void SqlTable::UpdateSchema(const catalog::Schema &schema) {
   STORAGE_LOG_INFO("Update schema version: {}", uint32_t(schema.GetVersion()));
   TERRIER_ASSERT(tables_.find(schema.GetVersion()) == tables_.end(), "schema versions for an SQL table must be unique");
   const auto layout_and_map = StorageUtil::BlockLayoutFromSchema(schema);
+
+  InverseColumnMap inv_col_map;
+  for (auto iter : layout_and_map.second) {
```

Comment:
Would it be possible to put the creation of InverseColumnMap inside BlockLayoutFromSchema? You should be able to easily create InverseColumnMap while you are creating ColumnMap. I feel like that can save us one loop iteration.

Reply:
I'm not sure the performance penalty is that much of a problem at this point, but I do agree that `BlockLayoutFromSchema` seems like a more appropriate place to put the construction logic.  Since that would up the number of return values to three, how would you prefer we handle the return?

At first glance, it looks like this function is only being called by us during the creation of a DataTableVersion.  Maybe we should hoist all of that logic out to StorageUtil, rename the function to something like `DataTableVersionFromSchema` and return the object?

Reply:
It sounds reasonable to me to rename the function.

Reply:
I'll knock that out over the weekend, then.